cdap-docs/build.sh

```diff
@@ -460,18 +490,74 @@ function copy_docs_inner_level() {
 }
 
 function build_zip() {
+  local zip_target=${1}
+  local zip_extras=${2}
   set_project_path
-  make_zip ${1}
-}
-
-function zip_extras() {
-  if [[ "x${1}" == "x" ]]; then
-    return
+  set_version
+  if [ "x${1}" == "x" ]; then
+    ZIP_DIR_NAME="${PROJECT}-docs-${PROJECT_VERSION}"
+  else
+    ZIP_DIR_NAME="${PROJECT}-docs-${PROJECT_VERSION}-$1"
   fi
-  echo "Adding .htaccess file (404 file)"
-  rewrite ${SCRIPT_PATH}/${COMMON_SOURCE}/${HTACCESS} ${TARGET_PATH}/${PROJECT_VERSION}/.${HTACCESS} "<version>" "${PROJECT_VERSION}"
   cd ${TARGET_PATH}
-  zip -qr ${ZIP_DIR_NAME}.zip ${PROJECT_VERSION}/.${HTACCESS}
+  doc_change_py="${TARGET_PATH}/../tools/doc-change.py"
+
+  echo "Removing old directories and zips"
+  rm -rf ${TARGET_PATH}/base 
+  rm -rf ${TARGET_PATH}/current
+  rm -rf ${TARGET_PATH}/${PROJECT_VERSION}
+  rm -rf ${TARGET_PATH}/*.zip
+
+  echo "Creating base"
+  mkdir base
+  cp -r ${HTML} base/en
+
+  # Add a redirect index.html file
+  echo "${REDIRECT_EN_HTML}" > base/index.html
+
+  echo
+  echo "1: 'current' version, with canonical ref link ${ZIP_DIR_NAME}-current"
+  cp -r base current
+  python ${doc_change_py} ${TARGET_PATH} --current=${PROJECT_VERSION} current
+  zip -qr ${ZIP_DIR_NAME}.zip current/* --exclude *.DS_Store* *.buildinfo*
+  if [[ "x${zip_extras}" != "x" ]]; then
+    echo "Creating and adding a .htaccess file (404 file)"
+    rewrite ${SCRIPT_PATH}/${COMMON_SOURCE}/${HTACCESS} ${TARGET_PATH}/current/.${HTACCESS} "<version>" "current"
+    zip -qr ${ZIP_DIR_NAME}.zip current/.${HTACCESS}
+  fi
+  mv ${ZIP_DIR_NAME}.zip ${ZIP_DIR_NAME}-current.zip
```

Comment:
There's no error checking in any of this. If any operation fails, it continues through the rest of the function. This function will only fail if the very last `rm` fails. This is used in our builds, so everything should fail fast, except for tests.